relays/bin-substrate/src/chains/millau.rs

```diff
@@ -25,24 +25,27 @@ use crate::cli::{
 };
 use anyhow::anyhow;
 use bp_message_dispatch::{CallOrigin, MessagePayload};
+use bp_runtime::EncodedOrDecodedCall;
 use codec::Decode;
 use frame_support::weights::{DispatchInfo, GetDispatchInfo};
 use relay_millau_client::Millau;
 use sp_version::RuntimeVersion;
 
 impl CliEncodeCall for Millau {
-	fn encode_call(call: &Call) -> anyhow::Result<Self::Call> {
+	fn encode_call(call: &Call) -> anyhow::Result<EncodedOrDecodedCall<Self::Call>> {
 		Ok(match call {
-			Call::Raw { data } => Decode::decode(&mut &*data.0)?,
+			Call::Raw { data } => Self::Call::decode(&mut &*data.0)?.into(),
```

Comment:
I'm confused, for Millau the same trait `fn encode_call()` actually decodes the call when raw data is provided?

Why not use `EncodedOrDecodedCall::Encoded(data.0.clone())` variant like in Kusama?

Reply:
This can be changed. But:
1) imo earlier error = better error. Since `Call` of Rialto/Millau is the runtime call (not the mocked call as for the Polkadot/Kusama), decode only fail if the call is invalid. It is better to see decoding error here, than some strange error later (e.g. on the bridged chain during dispatch);
2) most of times (except when sending messages manually or using sending max-dispatch-weight messages) in Rialto<>Millau bridge, we'll be calling `get_dispatch_info()` on this call to get its dispatch weight. So it needs to be decoded anyways.

Feel free to insist, if you think it'll be better to change it to the `Encoded` here, though :)